commands/sync/sync.sh

```diff
@@ -5,6 +5,31 @@ shopt -s inherit_errexit
 
 . "$(dirname "${BASH_SOURCE[0]}")/../cmd_runner.sh"
 
+# Function to check syncing status
+check_syncing() {
+  # Send the system_health request and parse the isSyncing field
+  RESPONSE=$(curl --silent --request POST http://127.0.0.1:9944 \
+    --header 'Content-Type: application/json' \
+    --data-raw '{"jsonrpc": "2.0", "method": "system_health", "params": [], "id": "1"}')
+
+  # Check for errors in the curl command
+  if [ $? -ne 0 ]; then
```

Comment:
Have you checked if you can actually reach this? `inherit_errexit` option, that's set on line 4, should do the trick automatically, like raising the exception. 

Here's an example script
```bashset -eu -o pipefail
shopt -s inherit_errexit

check() {
	result="$(curl -sS zadfdf.com)"
	if [ $? -ne 0 ]; then
		echo "WAAAT"
	fi
}

res=$(check)
```

And here's the execution
```text

cornholio-osx3:~/>bash -x a.bash                                                                
+ set -eu -o pipefail
+ shopt -s inherit_errexit
++ check
+++ curl -sS zadfdf.com
curl: (6) Could not resolve host: zadfdf.com
++ result=
+ res=
```

Reply:
@mutantcornholio yea it caught it when the node failed to run 
<img width="994" alt="image" src="https://github.com/paritytech/command-bot-scripts/assets/1177472/4aef05e1-08c5-4198-99d3-88ee97ce46a5">

Reply:
That's really weird. I even tried the script on `paritytech/ci-unified:bullseye-1.73.0-2023-05-23` to see if the bash version supports it, and still got the same result. Wonder what's it about..

Reply:
yea, you're right, i tried to reproduce locally and figured that it was showing error actually because of `set -x`, but in fact it was exiting successfully from job while was failing
I removed `shopt -s inherit_errexit` and refactored the while loop so it works a bit more explicit with catching results
Thank you @mutantcornholio !